### ISSUE TITLE
Opened as: "Grid List," meant: "Data Table"

### ISSUE THREAD

**Issue by ollwenjones** [justindujardin/ng2-material#31]
I'm likely to start using these components, but I'll need a grid-list pretty shortly.

I actually started out working with Material Design Lite CSS for some ng2 components, but I thought it would be better to contribute to something like this using the Angular Material CSS / methodology / designs.

Let me know if you have any guidance.

**Comment by DennisSmolek** [comment]
@ollwenjones, I never used grid-lists with ng1, what functionality are you looking for? There's some pretty cool stuff with flex going on that will do much of the box size, gutter, etc.

**Comment by ollwenjones** [comment]
App is going to have lots of tabular data for operators to interact with, so it's a honest to goodness data-grid, (not for layout).

I wish we didn't present the users with so much data in that form, but we're stuck having to give them a lot 'at a glance.'

**Comment by justindujardin** [comment]
There is a GridList component that exists in the angular/angular repo that I removed because it throws exceptions in debug mode (changeAfterChecked).

If someone were willing to spend time debugging the component, I can resurrect it and the example code in a branch.

**Comment by ollwenjones** [comment]
I'm hoping to spend some time on it this week, actually, which is really why I brought it up. ;) Starting the one in the angular/angular repo is pretty good guidance, thanks.

**Comment by ollwenjones** [comment]
I realized looking at ng-material today (finally) that "grid list" didn't mean what I meant. It's still more for layout than what I had in mind. What I had in mind was a more complex version of this:
http://www.getmdl.io/components/#tables-section

So maybe this issue should have been named something like "data tables" not "grid list"

I also didn't realize that ng-material didn't have a layout grid stuff that [mdl has](http://www.getmdl.io/components/#layout-section/grid) ... unless I'm missing it...

**Comment by ollwenjones** [comment]
@justindujardin Are you even interested in having something like http://www.google.com/design/spec/components/data-tables.html as part of this library.?
Those were really what I was after when I opened this issue. Sorry for the confusion.

**Comment by szechyjs** [comment]
I would love to have a component for the spec data tables.

**Comment by ollwenjones** [comment]
I'll need to take this on in some capacity pretty quick here... I know I said that about a month ago, haha.

**Comment by justindujardin** [comment]
@ollwenjones awesome, let me know what you come up with!

**Comment by Gregcop1** [comment]
Maybe this PR #117 can help you

**Comment by ollwenjones** [comment]
So there was a long discussion on PR #117 about data-table-component design, since @Gregcop1 and I took very different approaches.

Apologies @Gregcop1 for hijacking his PR with that discussion, which should have been happening here. :disappointed_relieved: 

I couldn't stop thinking about this last night, and I've had a bit of a change of heart. I thought of some other use cases I couldn't think of good flexible, encapsulated solutions for. A library like this has to provide the full-flexibility *directive* building blocks, but should it also provide a *component* that encapsulates them all for the 80%-90% use case?

Do we get behind @daniel-nagy's ng2 implementation, or just draw from his experience in what we build here?

TL;DR thought process details [in this gist](https://gist.github.com/ollwenjones/859433089782bebb8bee) if you're interested. Also if you're feeling generous, I appreciate any perspective on the technical challenges discussed therein. :wink: 

//cc @d3viant0ne, @justindujardin, @jraadt

**Comment by jraadt** [comment]
My vote would be to get behind @daniel-nagy's implementation.  At least use his ng1 version as a template or guide to start building a ng2 version.  He's done a great job putting together a data table for ng1.

**Comment by ollwenjones** [comment]
Going to close this issue since it's mostly about the existence of a data-table, which materially exists now (no pun intended.)

Further features / design discussion can happen in other issues.
